### PR TITLE
Add FXIOS-8582 Splash Screen Experiment

### PR DESCRIPTION
**Pull request by cyndichin**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8582)

## :bulb: Description
Phase I: Add animation for splash screen and configure feature flag and duration using Nimbus.

In terms of accessibility, still waiting design on confirmation and will update in a future PR, for now only show animation if Reduce Motion setting is disabled.

_Note: 
There is a delay for the animation due to some setup we are doing before showing the animation. Additionally, there is a bug in which the systemBackground color changes if the user was previously in private mode and launches from light mode.  These will be investigated in future work when we move Nimbus configuration._

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [x] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

## Screenshots
https://github.com/mozilla-mobile/firefox-ios/assets/6743397/d1f5d5f2-ef28-4c95-a82c-e2ca4dc92c1d

### CODE REVIEW

File: firefox-ios/Client/Nimbus/NimbusSplashScreenFeatureLayer.swift [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+protocol NimbusSplashScreenFeatureLayerProtocol {
```
Comment (cyndichin) [inline firefox-ios/Client/Nimbus/NimbusSplashScreenFeatureLayer.swift:7]: I followed what was done for the other features like `NimbusFirefoxSuggestFeatureLayerProtocol`. I also saw `NimbusFeatureFlagWithCustomOptionsID` having `searchBarPosition`, but felt that this way was cleaner than adding to that enum. 

Let me know if there's a better way in accessing this data, open to suggestions!

Comment (adudenamedruby) [reply firefox-ios/Client/Nimbus/NimbusSplashScreenFeatureLayer.swift:7]: No, this is the best way to do it. We should have layers for the different things. :)

**Review by adudenamedruby — APPROVED** [review]

File: firefox-ios/Client/Coordinators/Scene/SceneSetupHelper.swift [hunk 2]
```diff
@@ -26,10 +26,16 @@ struct SceneSetupHelper {
...
-        let navigationController = UINavigationController()
+        let navigationController = RootNavigationController()
         navigationController.isNavigationBarHidden = true
         navigationController.edgesForExtendedLayout = UIRectEdge(rawValue: 0)
 
         return navigationController
     }
 }
+
+class RootNavigationController: UINavigationController {
```
Comment (lmarceau) [inline firefox-ios/Client/Coordinators/Scene/SceneSetupHelper.swift:37]: nit; Should this go in its own file?

Comment (cyndichin) [reply firefox-ios/Client/Coordinators/Scene/SceneSetupHelper.swift:37]: curious on what's the team opinion on creating new files? For me its easy if everything is in one area without having to jump around if it's small, but happy to always create a new file for anytime a class or struct if that is preferred.

Comment (cyndichin) [reply firefox-ios/Client/Coordinators/Scene/SceneSetupHelper.swift:37]: discussed on Slack, will leave as is here, but going forward will keep in mind to create new files for struct / classes.

**Comment by cyndichin** [comment]
Closing and re-opening to retrigger Bitrise.

**Pull request merged**